modules/core/include/opencv2/core/hal/intrin_neon.hpp

```diff
@@ -1206,6 +1207,16 @@ inline v_float16x4 v_cvt_f16(const v_float32x4& a)
 }
 #endif
 
+//! @name Check SIMD support
+//! @{
+//! @brief Check CPU capability of SIMD operation
+static inline bool hasSIMD128()
```

Comment:
@vpisarev I believe we should allow to disabling optimizations during runtime (so we should call `checkHardwareSupport()`).